ligo/stablecoin/actions.ligo

```diff
@@ -0,0 +1,288 @@
+(* SPDX-FileCopyrightText: 2020 tqtezos
+ * SPDX-License-Identifier: MIT *)
+
+(*
+ * Root entrypoints for stablecoin smart-contract
+ *)
+
+#include "operator.ligo"
+
+(* ------------------------------------------------------------- *)
+
+(*
+ * Helpers and validation
+ *)
+
+(*
+ * This function fails if provided token_id is not equal to
+ * default one, restricting all operations to be monolitic
+ *)
+function validate_token_type
+  ( const token_id : token_id
+  ) : unit is
+    if token_id =/= default_token_id
+    then failwith ("This operation does not support multiple token types")
+    else unit
+
+(*
+ * Same as above but for a list of token ids
+ *)
+function validate_token_types
+  ( const token_ids : list (token_id)
+  ) : unit is List.fold
+    ( function
+        ( const u        : unit
+        ; const token_id : token_id
+        ) : unit is validate_token_type (token_id)
+    , token_ids
+    , unit
+    )
+
+(*
+ * Increase the balance of an address associated with it
+ * and present in ledger. If not, create a new address with
+ * that balance given.
+ *)
+function credit_to
+  ( const parameter : mint_params
+  ; const store     : storage
+  ) : storage is block
+{ var updated_balance : nat := 0n
```

Comment:
Not sure but wouldn't it be less error prone to not use a `var` here? And instead use a `const` to hold the existing balance and the updated balance?